manifests/init.pp

```diff
@@ -1,20 +1,49 @@
+# == Class: homebrew
+#
+# This class is able to install and configure homebrew and brew cask on node.
+#
+# === Parameters
+#
+# [*user*]
+#   User owner of the brew files and folders.
+#   Defaults to nothing.
+#
+# [*group*]
+#   Group owner of the brew files
+#   Defaults to `undef`.
+#
+# [*command_line_tools_package*]
+#   Url where to download the command line tools package.
+#   Defaults to `undef`.
+#
+# [*command_line_tools_source*]
+#   name of the dmg file containing the installed for the command line tools package.
+#   Defaults to `undef`.
+#
+# === Original Author
+# Original credit for this module goes to
+# [kelseyhightower](https://github.com/kelseyhightower). This module was forked
+# to provide brewcask integration.
+#
+# === Contributers
+# Kevin James <KevinJames@thekev.in>
+# Jordi Garcia <jordi@jordigarcia.net>
+#
+# === Copyright
+#
+# Copyright 2015-Present Kevin James, unless otherwise noted.
+#
+# Credit for logic involved in tapping repositories goes to
+# [gildas](https://github.com/gildas/puppet-homebrew).
+#
 class homebrew (
-  $user,
-  $command_line_tools_package = undef,
-  $command_line_tools_source  = undef,
-  $group = 'admin'
-) {
-
-  if $::operatingsystem != 'Darwin' {
-    err('This Module works on Mac OS X only!')
-    fail('Exit')
-  }
-
-  if $homebrew::user == 'root' {
-    warning('Homebrew will be dropping support for root-owned brew by November 2016. Though this module will not prevent you from installing homebrew as root, you may run into unexpected issues. It is highly recommended you follow brew guidelines (install as a non-root user) -- this module will enforce this once homebrew has officially dropped support for root-owned installations.')
-  }
-
-  include homebrew::compiler
-  include homebrew::install
-
+  $user = $brew::params::user,
+  $group = $brew::params::group,
+  $command_line_tools_package = $brew::params::command_line_tools_package,
+  $command_line_tools_source  = $brew::params::command_line_tools_source
+) inherits ::homebrew::params {
```

Comment:
Inheriting from `params.pp`: no. Couple reasons for this: [lint issues](http://puppet-lint.com/checks/class_inherits_from_params_class/) and you changed how the user variable works, allowing user to be unset. We don't want that.